src/software/world/ball_state_with_timestamp.cpp

```diff
@@ -0,0 +1,34 @@
+#include "software/world/ball_state_with_timestamp.h"
+
+BallStateWithTimestamp::BallStateWithTimestamp(const Point &position,
+                                               const Vector &velocity,
+                                               const Timestamp &timestamp)
+    : BallState(position, velocity), timestamp_(timestamp)
+{
+}
+
+BallStateWithTimestamp::BallStateWithTimestamp(const BallState &ball_state,
+                                               const Timestamp &timestamp)
+    : BallState(ball_state), timestamp_(timestamp)
+{
+}
+
+Timestamp BallStateWithTimestamp::timestamp() const
+{
+    return timestamp_;
+}
+
+BallState BallStateWithTimestamp::getBallState() const
+{
+    return BallState(position_, velocity_);
+}
+
+bool BallStateWithTimestamp::operator==(const BallStateWithTimestamp &other) const
+{
+    return this->getBallState() == other.getBallState();
```

Comment:
We should compare timestamps as well no? We should definitely add a unit test for this case.

Reply:
Yes but I didn't want to change the spec of the function (it already wasn't comparing timestamps). I agree this is bad, and have created #1356 to deal with that interface change separately in case it cascades into more changes.